shoop/themes/classic_gray/views/product_preview.py

```diff
@@ -8,7 +8,21 @@
 from shoop.front.views.product import ProductDetailView
 
 
+class CGProductDetailView(ProductDetailView):
+    def get_context_data(self, **kwargs):
+        # By default the template rendering the basket add form
+        # uses the `request.path` as its' `next` value.
+        # This is fine if you are on product page but here in
+        # preview, we cannot redirect back to `/xtheme/product_preview`.
+
+        context = super(CGProductDetailView, self).get_context_data(**kwargs)
+        # Add `return_url` to context to avoid usage of `request.path` in
```

Comment:
Aha! Okay, the class being named `CGProductDetailView`, not `ProductPreviewView` or something like that threw me off.

So could you rename the class to `ProductPreviewView`? You could then also move the `template_name="classic_gray/product_preview.jinja"` override from `product_preview()` into the class body.